### PR TITLE
Ci testing

### PR DESCRIPTION
**Pull request by BenediktSeidl**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -62,6 +62,31 @@ jobs:
...
+      - name: Setup Go environment
+        uses: actions/setup-go@v3
+        with:
+          go-version: '1.19'
+
+      - name: Install validator
+        run: |
+          git clone https://github.com/grafana/plugin-validator
+          pushd ./plugin-validator/pkg/cmd/plugincheck2
+          go install
```
Comment (TheSpacyCat) [inline .github/workflows/ci.yml:89]: We can use a binary from one of the releases. The tars contain an executable file, a README and a LICENSE, nothing else.

File: README.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # Checkmk's Grafana Data Source Plugin
 
-![CI](https://github.com/checmk/grafana-checkmk-datasource/actions/workflows/ci.yml/badge.svg)
+![CI](https://github.com/Checkmk/grafana-checkmk-datasource/actions/workflows/ci.yml/badge.svg)
```
Comment (TheSpacyCat) [inline README.md:3]: Ok, I'll have to rewrite these everywhere else as well. I'll put it on my TODO list.

File: .github/build/action.yml [hunk 3]
```diff
@@ -0,0 +1,76 @@
+name: 'Build'
+inputs:
+  build_identifier:
+    description: 'If specified will be added to the version in order to identify non official releases'
+    required: false
+    default: ''
+outputs:
+  changelog_outputs_path:
```
Comment (TheSpacyCat) [inline .github/build/action.yml]: Maybe we'll give these nice names in the future, e.g. changelog, archive, checksum

File: .github/build/action.yml [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    value: '${{ steps.metadata.outputs.archive }}'
+  metadata_outputs_archive-checksum:
+    description: 'archive-checksum'
+    value: '${{ steps.metadata.outputs.archive-checksum }}'
+runs:
+  using: 'composite'
+  steps:
+    - name: Install dependencies
+      shell: bash
+      run: yarn install --frozen-lockfile
```
Comment (TheSpacyCat) [inline .github/build/action.yml]: Just out of interest: What is the difference between '--immutable' and '--forzen-lockfile'

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -32,17 +32,21 @@ jobs:
...
 
       - name: Compatibility check
         run: npx @grafana/levitate@latest is-compatible --path src/module.ts --target @grafana/data,@grafana/ui,@grafana/runtime
 
-      - name: Upload package for e2e tests
+      - name: Build frontend
+        id: build
+        uses: ./.github/build/
+        with:
+          build_identifier: dev.${{ github.run_id }}
```
Comment (TheSpacyCat) [inline .github/workflows/ci.yml]: Have you thought about using a short commit sha here?

File: .github/workflows/release.yml [hunk 6]
```diff
@@ -76,9 +37,9 @@ jobs:
...
           tag: ${{ github.ref }}
           name: Release ${{ github.ref }}
-          body: ${{ steps.changelog.outputs.path }}
+          body: ${{ steps.build.changelog_outputs_path }}
           draft: true
-          artifacts: ./${{ steps.metadata.outputs.archive }}
+          artifacts: ./${{ steps.build.metadata_outputs_archive }}
           artifactContentType: application/zip
 
       - name: Add checksum to release
```
Comment (TheSpacyCat) [inline .github/workflows/release.yml]: We can put that into the step above.

File: .github/workflows/release.yml [hunk 7]
```diff
@@ -76,9 +37,9 @@ jobs:
         with:
           tag: ${{ github.ref }}
           name: Release ${{ github.ref }}
-          body: ${{ steps.changelog.outputs.path }}
+          body: ${{ steps.build.changelog_outputs_path }}
           draft: true
-          artifacts: ./${{ steps.metadata.outputs.archive }}
+          artifacts: ./${{ steps.build.metadata_outputs_archive }}
```
Comment (TheSpacyCat) [inline .github/workflows/release.yml]: That one can be changed to './*zip' to include a zip for free and cloud.

File: .github/build/action.yml [hunk 8]
```diff
@@ -70,6 +99,9 @@ runs:
       shell: bash
       id: package-plugin
       run: |
+        # clean up previous build # TODO: think about better solution
```
Comment (TheSpacyCat) [inline .github/build/action.yml]: We could check out the repo two times and run the cloud build in one path and the free build in the other.

File: README.md (hunk 2 above)
Comment (BenediktSeidl) [reply README.md:3]: are you sure? i can not finy any matches for this typo.

File: .github/build/action.yml (hunk 4 above)
Comment (BenediktSeidl) [reply .github/build/action.yml]: i have no idea. this was c&p from the grafana provided workflow. but they changed from yarn to npm and in the current workflows, the both use `npm ci` already added the switch to npm on our long term todo

File: .github/workflows/ci.yml (hunk 5 above)
Comment (BenediktSeidl) [reply .github/workflows/ci.yml]: we can if you want. but you can build twice for one commit, and the run id will increment (i hope)

File: README.md (hunk 2 above)
Comment (TheSpacyCat) [reply README.md:3]: Found a few other ones in the plugin.json. I'll do a fixup MR

File: .github/build/action.yml (hunk 8 above)
Comment (BenediktSeidl) [reply .github/build/action.yml]: adapted this one of the commits

File: .github/build/action.yml (hunk 3 above)
Comment (BenediktSeidl) [reply .github/build/action.yml]: should be done

File: .github/build/action.yml (hunk 4 above)
Comment (TheSpacyCat) [reply .github/build/action.yml]: fine for me

File: .github/workflows/ci.yml (hunk 1 above)
Comment (BenediktSeidl) [reply .github/workflows/ci.yml:89]: this was c&p from the offical actions, but you are right, i adapted it.

File: .github/workflows/ci.yml (hunk 5 above)
Comment (TheSpacyCat) [reply .github/workflows/ci.yml]: fine for me.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (TheSpacyCat) [reply .github/workflows/ci.yml:89]: fine for me

File: .github/workflows/release.yml (hunk 6 above)
Comment (BenediktSeidl) [reply .github/workflows/release.yml]: but then they have both `artifactContentType: raw`. we will see if this is a problem

File: .github/workflows/release.yml (hunk 7 above)
Comment (BenediktSeidl) [reply .github/workflows/release.yml]: i think it's good to be specific here, i would hope we can notice problems earlier this way.

File: .github/workflows/ci.yml (hunk 5 above)
Comment (BenediktSeidl) [reply .github/workflows/ci.yml]: would keept it as is, but we can adapt later on if you want.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (BenediktSeidl) [reply .github/workflows/ci.yml:89]: does not work, as this file `./plugin-validator/config/default.yaml` is missing. i'm going to remove this commit. we can look into this later

**Pull request merged**